common/src/crypto.rs

```diff
@@ -11,13 +11,103 @@ Abstract:
     Crypto helper routines
 
 --*/
-use caliptra_drivers::{Ecc384PubKey, KeyId};
-/// DICE  Layer Key Pair
-#[derive(Debug)]
+use caliptra_drivers::{
+    Array4x12, Array4x8, CaliptraResult, Ecc384PubKey, Hmac, HmacMode, KeyId, KeyReadArgs,
+    KeyUsage, KeyWriteArgs, Mldsa87PubKey, Sha256, Sha256Alg, Sha2_512_384, Trng,
+};
+use zeroize::Zeroize;
+
+/// DICE Layer ECC Key Pair
+#[derive(Debug, Zeroize)]
 pub struct Ecc384KeyPair {
-    /// Private Key
+    /// Private Key KV Slot Id
+    #[zeroize(skip)]
     pub priv_key: KeyId,
 
     /// Public Key
     pub pub_key: Ecc384PubKey,
 }
+
+/// DICE Layer MLDSA Key Pair
+#[derive(Debug, Zeroize)]
+pub struct MlDsaKeyPair {
+    /// Key Pair Generation KV Slot Id
+    #[zeroize(skip)]
+    pub key_pair_seed: KeyId,
+
+    /// Public Key
+    pub pub_key: Mldsa87PubKey,
+}
+
+#[derive(Debug)]
+pub enum PubKey<'a> {
+    Ecc(&'a Ecc384PubKey),
+    Mldsa(&'a Mldsa87PubKey),
+}
+
+/// Calculate SHA2-256 Digest
+///
+/// # Arguments
+///
+/// * `sha256` - SHA256 driver
+/// * `data`   - Input data to hash
+///
+/// # Returns
+///
+/// * `Array4x8` - Digest
+pub fn sha256_digest(sha256: &mut Sha256, data: &[u8]) -> CaliptraResult<Array4x8> {
+    sha256.digest(data)
```

Comment:
I'm not sure why we need this function, which just wraps a simple method call? (ditto for sha384)

Reply:
Updated.